Integrate Firebase authentication in dashboard app

DillanPillai opened the pull request:
Refactored frontend authentication to use Firebase Auth, including new context, provider, and hooks. Added firebase config loader, updated ProtectedRoute for redirect logic, and improved login page to handle Firebase-based login and redirects. Updated dependencies to include Firebase packages. Minor code cleanup and improved type safety.

DillanPillai committed:
Integrate Firebase authentication in dashboard app

Refactored frontend authentication to use Firebase Auth, including new context, provider, and hooks. Added firebase config loader, updated ProtectedRoute for redirect logic, and improved login page to handle Firebase-based login and redirects. Updated dependencies to include Firebase packages. Minor code cleanup and improved type safety.

Kaurnavreet1699 reviewed (commented):
There’s an issue in `frontend/dashboard/src/main.tsx` — the new `AuthProvider` and `BrowserRouter` wrapper block is added **below** the render call, so it won’t actually execute. Only the original `createRoot(...).render(<App />)` line is used.  

The correct structure should be to wrap `<App />` with `AuthProvider` and `BrowserRouter` directly in the render call, like this:

```tsx
import { createRoot } from 'react-dom/client'
import App from './App.tsx'
import './index.css'
import { AuthProvider } from './contexts/AuthContext.tsx'
import { BrowserRouter } from 'react-router-dom'

createRoot(document.getElementById("root")!).render(
  <AuthProvider>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </AuthProvider>
)

DillanPillai committed:
Wrap App component with BrowserRouter

The App component is now wrapped with BrowserRouter to enable routing functionality. This change ensures that React Router features are available throughout the application.

DillanPillai commented:
Issue resolved

Kaurnavreet1699 reviewed (approved):
Looks good overall 👍 The Firebase Auth integration, context setup, and improved routing logic are well-structured. Thanks for cleaning up types and dependencies too this makes the codebase much clearer. Approving merge
